Bug 1389176: fix issue #36: Add i18n to SMS messages

pmac opened the pull request:
* Add support for different countries and languages to basket SMS admin
* Add support for parsing and formatting international phone numbers via the python port of libphonenumber from Google

**NOTE**: This should be deployed in three steps, one for each commit in this PR. So the commits should **NOT** be squashed and should be tagged and deployed separately. This will ensure that the code and data migrations make it to prod in the proper order and not cause errors.

glogiotatidis commented on basket/news/views.py:
```diff
@@ -464,30 +464,30 @@ def invalid_email_response():
+    if not vendor_id:
         return HttpResponseJSON({
             'status': 'error',
             'desc': 'Invalid msg_name',
```
Change error message to `invalid msg_name, country, language combination`?

glogiotatidis commented on basket/news/models.py:
```diff
@@ -308,9 +307,21 @@ class SMSMessage(models.Model):
     class Meta:
-        verbose_name = "SMS message"
+        unique_together = ['message_id', 'language', 'country']
+        verbose_name = "Localized SMS message"
```
nit: `s/"/'/g`

glogiotatidis commented on basket/news/newsletters.py:
```diff
@@ -8,21 +8,16 @@
 
 CACHE_KEY = "newsletters_cache_data"
-SMS_CACHE_KEY = "sms_messages_cache_data"
+SMS_CACHE_KEY = "local_sms_messages_cache_data"
```
nit: `s/"/'/g`

glogiotatidis commented on basket/news/newsletters.py:
```diff
@@ -46,15 +41,21 @@ def get_sms_messages():
     data = cache.get(SMS_CACHE_KEY)
     if data is None:
         # TODO have this be an empty dict when SMS_MESSAGES is removed.
```
Remove comment?

glogiotatidis commented on requirements/base.txt:
```diff
@@ -132,3 +132,6 @@ APScheduler==3.3.1 \
     --hash=sha256:f68874dff1bdffcc6ce3adb7840c1e4d162c609a3e3f831351df30b75732767b
 tzlocal==1.3.2 \
     --hash=sha256:07dee68c9f5b9ca665762eb0860418cc289bd561f7e32a4ec37a09426895e7bd
+phonenumbers==8.8.1 \
```
nice library 👍

pmac commented:
@glogiotatidis thanks! I've updated the things you mentioned. I also made a change that was just bugging me. I get the `vendor_id` for the SMS message in the view, but then just drop it and send the message ID, country, and language via the queue. I have to only add optional kwargs to the task since old tasks with the old function signature will be on the queue when it's deployed. So instead I've added `vendor_id` as the optional kwarg and will use that instead of discovering it again in the task. This seems cleaner to me. I'll follow up with more changes to the signature to clean it up.

glogiotatidis commented:
Perfect! ship it!
